performance_regression_test.py

```diff
@@ -790,3 +791,114 @@ def build_histogram(self, workload: PerformanceTestWorkload, test_type: Performa
+        with EventsSeverityChangerFilter(new_severity=Severity.NORMAL,  # killing stress creates Critical error
+                                         event_class=CassandraStressEvent,
+                                         extra_time_to_expiration=60):
+            self.loaders.kill_stress_thread()
+
+    @latency_calculator_decorator
+    def steady_state_latency(self):  # pylint: disable=no-self-use
+        sleep_time = self.db_cluster.params.get('nemesis_interval') * 60
+        InfoEvent(message='Starting Steady State calculation for %ss' % sleep_time).publish()
+        time.sleep(sleep_time)
+        InfoEvent(message='Ended Steady State calculation. Took %ss' % sleep_time).publish()
+
+    @latency_calculator_decorator
+    def post_upgrades_steady_state(self):
+        sleep_time = self.db_cluster.params.get('nemesis_interval') * 60
+        InfoEvent(message='Starting Post-Upgrade Steady State calculation for %ss' % sleep_time).publish()
+        time.sleep(sleep_time)
+        InfoEvent(message='Ended Post-Upgrade Steady State calculation. Took %ss' % sleep_time).publish()
+
+    def run_workload_and_upgrade(self, stress_cmd, sub_type=None):
+        # workaround to have it working inside `latency_calculator_decorator`
+        self.cluster = self.db_cluster  # pylint: disable=attribute-defined-outside-init
+        # workaround to have it working inside `latency_calculator_decorator`
+        self.tester = self  # pylint: disable=attribute-defined-outside-init
+        # workaround to have it working inside `latency_calculator_decorator`
+        self.monitoring_set = self.monitors  # pylint: disable=attribute-defined-outside-init
+        if sub_type is None:
+            sub_type = 'read' if ' read ' in stress_cmd else 'write' if ' write ' in stress_cmd else 'mixed'
+        test_index = f'latency-during-upgrade-{sub_type}'
+        self.create_test_stats(sub_type=sub_type, append_sub_test_to_name=False, test_index=test_index)
+        stress_queue = self.run_stress_thread(stress_cmd=stress_cmd, stress_num=1, stats_aggregate_cmds=False)
+        self.steady_state_latency()
+        versions_list = []
+
+        def _get_version_and_build_id_from_node(node):
+            version = node.remoter.run('scylla --version')
+            build_id = node.remoter.run('scylla --build-id')
+            return version.stdout.strip(), build_id.stdout.strip()
+
+        for node in self.db_cluster.nodes:
+            base_version, base_build_id = _get_version_and_build_id_from_node(node)
+            self._upgrade_node(node)
+            target_version, target_build_id = _get_version_and_build_id_from_node(node)
+            versions_list.append({'base_version': base_version,
+                                  'base_build_id': base_build_id,
+                                  'target_version': target_version,
+                                  'target_build_id': target_build_id
+                                  })
```

Comment:
i suggest here to add also node name, so in any failure it will easy to see which node were upgraded or not

Reply:
@aleksbykov , added